packages/firebase_database/ios/Classes/FirebaseDatabasePlugin.m

```diff
@@ -7,7 +7,9 @@
 #import <Firebase/Firebase.h>
 
 static FlutterError *getFlutterError(NSError *error) {
-  return [FlutterError errorWithCode:[NSString stringWithFormat:@"Error %d", (int)error.code]
+  if (error == nil) return nil;
```

Comment:
I personally prefer the idea to add the check before where the function is called like everywhere else. It avoids hiding a secret nil return. Also it keeps the consistency throughout the repo. Same for the cloud_firestore plugin.

Reply:
I also prefer doing the check on the parent. However, when fixing code, I'd rather change it as little as possible to avoid new errors.

Reply:
fair enough!